### PR TITLE
Add script for identifying master runs that are missing tags

### PR DESCRIPTION
**Pull request by stephenmcgruer**

### CODE REVIEW

**Comment by stephenmcgruer** [comment]
@Hexcles, @foolip - PTAL. This is what I have atm for 2017 runs that I am considering tagging as 'master' (which I would do via a data-migration script as normal). There are some considerations here however:

- Although most have ~24k entries, diffing `git show ... --stat` of different 24k runs for the same browser (Firefox was what I tested) had a ~2.5k different entries, with no obvious pattern.
- There is one 30k entry, for no obvious reason as far as I can tell.

Given that https://wpt.fyi/api/runs?label=master&to=2017-12-31T00:00:00Z is currently empty however, and the only thing we would really use this data for is historical graphing, I think this is probably still a reasonable set to tag?

Actually, one thing I haven't done is check that the SHAs involved are in web-platform-tests/wpt.git. Would we expect that all master runs, even 2017 historical ones, are tagged with a SHA that is in there?

Note: I don't intend to land this script, hence why its still draft.

**Comment by foolip** [comment]
@stephenmcgruer I'll review this now, but I'd recommend either tackling https://github.com/web-platform-tests/wpt.fyi/issues/1540 before using the script or deciding to never salvage those runs, so that we don't have to re-run this script after salvaging them.

**Comment by foolip** [comment]
> Actually, one thing I haven't done is check that the SHAs involved are in web-platform-tests/wpt.git.
> Would we expect that all master runs, even 2017 historical ones, are tagged with a SHA that is in there?

Yes, there could be runs from branches that were later merged into master and commits that aren't reachable from wpt master at all. Neither should be tagged as master.

There are merge_pr_* tags from 2017-07-01T00:00Z for every commit that master ever pointed to, so I'd filter this list to the set of shas in `git tag --list 'merge_pr_*' --format="%(objectname)"`.

File: identify-master.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    if (diffList.length != 1) {
+      throw new Error(`${run} has diffList of length ${diffList.length} (expected 1)`);
+    }
+
+    // This is the slow bit. Takes about a second per run.
+    let stats = await diffList[0].getStats();
+    let insertions = stats.insertions();
+
+    // 2017 runs have a little over 24k subtests.
+    if (insertions >= 24000) {
```
Comment (foolip) [inline identify-master.js]: Can you list the number of tests for all candidate runs and sort by test count? If there are runs with just shy of 24k tests I'd be concerned that some of them with more are also incomplete.

Comment (stephenmcgruer) [reply identify-master.js]: Done; see 2017-runs-log.txt. Looks like there are runs with just shy of 24k, so you may be right.

**Comment by stephenmcgruer** [comment]
>  web-platform-tests/wpt.fyi#1540

This has been fixed now.

**Comment by stephenmcgruer** [comment]
403 new entries after web-platform-tests/wpt.fyi#1540, btw. Still to filter by `merge_pr_*` tags.

**Comment by stephenmcgruer** [comment]
> To figure out where to draw the line, I'd check a few against the manifests.

Checked with @Hexcles - we don't have data for manifests pre-2018. So for 2017 I'm going to go with what we have, drawing a line based on, well, my own judgement.

**Comment by stephenmcgruer** [comment]
2017 has been done, using the code in this PR plus a simple data migration script to check for a matching run id and add the 'master' label. 251 runs were tagged:

[2017-runs-tagged-master.txt](https://github.com/foolip/ad-hoc-wpt-results-analysis/files/3919127/2017-runs-tagged-master.txt)

I am now moving onto looking at 2018; I will open a new PR which will be basically the same code but changed for 2018, just to keep things clean.

**Pull request closed without merging**